code/datums/weather/weather_types/solar_flare.dm

```diff
@@ -35,9 +36,24 @@
 	// Solars produce 40x as much power. 240KW becomes 9.6MW. Enough to cause APCs to arc all over the station if >=2 solars are hotwired.
 	SSsun.solar_gen_rate = initial(SSsun.solar_gen_rate) * 40
 
+/datum/weather/solar_flare/can_weather_act(mob/living/L)
+	. = ..()
+	if(.) //If true the mob is already affected, no need to keep processing
+		return TRUE
+	if(istype(L, /mob/living/simple_animal)) //while this might break immersion, I don't want to spam the server with calling this on simplemobs
+		return FALSE
+	for(var/turf/T in oview(L))
```

Comment:
might want to do `oview(get_turf(L))` that way you don't get burnt for having X-ray vision.

Reply:
I think I'll leave it as is, while there is some merit to this suggestion I feel like it's a reasonable drawback to seeing through walls.

Reply:
yes